Sources/Sentry/SentryFileManager.m

```diff
@@ -634,23 +634,132 @@ - (BOOL)isDirectory:(NSString *)path
     return [NSFileManager.defaultManager fileExistsAtPath:path isDirectory:&isDir] && isDir;
 }
 
+/**
+ * @note This method must be statically accessible because it will be called during app launch,
+ * before any instance of  ``SentryFileManager`` exists, and so wouldn't be able to access this path
+ * from an objc property on it like the other paths.
+ */
 NSString *_Nullable sentryStaticCachesPath(void)
 {
     static NSString *_Nullable sentryStaticCachesPath = nil;
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
-        NSString *cachesDirectory
+        // We request the users cache directory from Foundation.
+        // For iOS apps and macOS apps with sandboxing, this path will be scoped for the current
+        // app. For macOS apps without sandboxing, this path is not scoped and will be shared
+        // between all apps.
+        NSString *_Nullable cachesDirectory
             = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES)
                   .firstObject;
         if (cachesDirectory == nil) {
             SENTRY_LOG_WARN(@"No caches directory location reported.");
             return;
         }
-        sentryStaticCachesPath = cachesDirectory;
+
+        // We need to ensure our own sandbox so that this path is not shared between other apps on
+        // the same system.
+        NSString *_Nullable sandboxedCachesDirectory
+            = sentryGetScopedCachesDirectory(cachesDirectory);
```

Comment:
`m`: This can be a bit confusing. On a non-sandboxed macOS app this directory isn't sandboxed. If you're not familiar with the code, you could assume we create our own sandbox, which isn't true. Instead, we use the public caches folder and scope the directory per app, so it isn't shared across apps. We explain this in detail below, but if you only read this section of the code, you could assume we create our own sandbox.
```suggestion
        // We need to ensure our own scoped directory so that this path is not shared between other apps on
        // the same system.
        NSString *_Nullable scopedCachesDirectory
            = sentryGetScopedCachesDirectory(cachesDirectory);
```

Reply:
I agree, left-over from refactoring